code/game/atoms_movable.dm

```diff
@@ -72,6 +72,8 @@
 	/// Whether this atom should have its dir automatically changed when it moves. Setting this to FALSE allows for things such as directional windows to retain dir on moving without snowflake code all of the place.
 	var/set_dir_on_move = TRUE
 
+	var/calculating_path = FALSE
```

Comment:
Document this.
Also, is this strictly necessary?
You can get away with having an indexed list, if the number of elements won't be too large, given this is adding a variable that will be changed (and thus defined) at runtime for every single movable atom.
Unless if you really think the perf of the list access is not worth it.

Reply:
Yeah I thought I might need this later on but it turns out nowhere in our code cares if we're currently generating a path, so I'll leave it for whoever adds a reason to check for that to add the proper check